### DISCUSSION TITLE
Breaking tps api?

### DISCUSSION THREAD

**Discussion by masmc05** [PurpurMC/Purpur#1069] [__Dismissed__]
https://github.com/PurpurMC/Purpur/blob/ver/1.19/patches/server/0067-Add-5-second-tps-average-in-tps.patch
can't this make plugins show incorrect tps, when they hard code that in index 0 is 1 minute tps, in 1 is 5 minutes etc. ?
https://github.com/PurpurMC/Purpur/blob/028780d4e124f3b3c81e9c0a28308b577e102f68/patches/server/0067-Add-5-second-tps-average-in-tps.patch#L79
this for sure looks like made to break plugins, very strange changes in api, where plugins can't check that index 1 is now desired tps instead of 0 etc.

is this really safe? shouldn't in api at least it be a separate call (like get5STps()) where plugins that would like to use 5 could use, and plugins which expect at index 0 to be 1 minute average, get 1 minute average? 

maybe you know better these changes and could explain better

**Comment by BillyGalbreath** [comment]
> this for sure looks like made to break plugins

You caught me. My entire motive was to break plugins.

**Reply by masmc05** [reply]
yeah, evil purpur XD
but anyways, maybe this patch could be remade in a way to avoid such possibilities that plugins will show wrong values?
